src/core/qgismobileapp.cpp

```diff
@@ -99,6 +101,7 @@ QgisMobileapp::QgisMobileapp( QgsApplication *app, QObject *parent )
   connect( this, &QQmlApplicationEngine::quit, app, &QgsApplication::quit );
 
   QgsQuickMapCanvasMap *mMapCanvas = rootObjects().first()->findChild<QgsQuickMapCanvasMap *>();
+  mMapCanvas->mapSettings()->setProject( mProject );
```

Comment:
@m-kuhn now that QgsQuickMapSettings has a setProject method, I used it here (successfully tested)